File: calculate_fCOS/get_C3C4_light_LRU_eq.R
library(lattice)
library(RColorBrewer)

##parse and pre-process data
fpath <- file.path(Sys.getenv('HOME'), 'work', 'Data',
                   'Stimler_COS_exchange_data.csv')
all_data <- read.csv(fpath)
all_data <- all_data[-1, c('PAR_umol.m.2s.1', 'lru', 'plant')]
names(all_data) <- c('PAR', 'LRU', 'species')
light_experiments <- grep('-light', as.character(all_data[['species']]))

light_data[['species']] <- as.factor(sub('-light', '',
                                         as.character(light_data[['species']])))
isC4 <- light_data[['species']] %in% c('corn', 'sorghum', 'amaranthus')
light_data[['C3C4']][isC4] <- 'C4'
light_data[['C3C4']][!isC4] <- 'C3'
light_data[['C3C4']] <- as.factor(light_data[['C3C4']])


pal <- brewer.pal(length(levels(light_data[['species']])), 'Dark2')
mytheme <- standard.theme("pdf")
mytheme[['superpose.symbol']][['pch']] <- c(15,16,17,3,4)
##mytheme[['superpose.line']][['col'] <- pal # doesn't affect the loess lines
mytheme[['superpose.symbol']][['col']] <- pal

xy <- xyplot(LRU~PAR|C3C4, data=light_data, groups=species,
             panel=panel.superpose,
             auto.key=TRUE,
             col.line=pal,
             par.settings=mytheme,
             panel.groups=function(x, y, ...){
                 panel.xyplot(x, y, ...)
                 panel.loess(x, y, ...)})
print(xy)

C3_lo <- loess(LRU~PAR, data=light_data[light_data[['C3C4']] == 'C3', ])
C4_lo <- loess(LRU~PAR, data=light_data[light_data[['C3C4']] == 'C4', ])

C3_pred <- predict(C3_lo, newdata=seq(0, max(light_data[['PAR']])))
C4_pred <- predict(C4_lo, newdata=seq(0, max(light_data[['PAR']])))

File: calculate_fCOS/get_C3C4_light_LRU_eq.py
import sys, os, os.path
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

C4_plants = ['corn-light', 'sorghum-light', 'amaranthus-light']
C3_plants = ['sage-light', 'hibiscus-light']
#light_experiments = C3_plants.append(C4_plants)

fpath = os.path.join(os.getenv('HOME'), 'work', 'Data',
                     'Stimler_COS_exchange_data.csv')
all_data = pd.read_csv(fpath, sep=',', skiprows=[1])
light_data = all_data.loc[:, ['PAR_umol m-2s-1', 'lru', 'plant']]
light_data = light_data.ix[all_data.plant.str.contains('(light)', na=False)]

# light_data.loc[:, ('C3C4')] = 'C3'
isC4 = light_data.plant.str.contains('(corn|sorghum|amaranthus)')
light_data.loc[isC4, ('C3C4')] = 'C4'
light_data.loc[~isC4, ('C3C4')] = 'C3'

light_data.columns = light_data.columns = [u'PAR', u'LRU', u'plant', u'C3C4']

sns.set_style('ticks')
sns.lmplot("PAR", "LRU", light_data, hue="C3C4", markers=["x", "o"], lowess=True)
g = sns.lmplot("PAR", "LRU", light_data, hue="plant", lowess=True, legend_out=False)
plt.show()
# g = sns.FacetGrid(light_data, hue="plant", size=5, 
#                   palette=sns.color_palette('Dark2'),
#                   legend_out=False)
# g.map(plt.scatter, "PAR", "LRU")
# g.add_legend();
# plt.show()
